src/nvim/tui/tui.c

```diff
@@ -521,6 +530,16 @@ static void update_attrs(UI *ui, HlAttrs attrs)
   if (italic) {
     unibi_out(ui, unibi_enter_italics_mode);
   }
+  if (undercurl || (underline && data->unibi_ext.enter_undercurl_mode)) {
```

Comment:
`underline && data->unibi_ext.enter_undercurl_mode` looks a bit weird. Is the set of terminals which support `4:3` code identical to set of terminals that supports `58` color? 

Regardless this could be changed to `(undercurl || underline) && data->unibi_ext.set_underline_color` (and `enter_undercurl_mode` check need not be nested) in case the detection code becomes more flexible (for terminals that support one but not the other). Also shouldn't `attrs.rgb_sp_color != -1` be checked?

Reply:
> Is the set of terminals which support 4:3 code identical to set of terminals that supports 58 color?

That's what I've assumed, but only based on https://sw.kovidgoyal.net/kitty/protocol-extensions.html?highlight=extensions#colored-and-styled-underlines.